stonesoup/architecture/__init__.py

```diff
@@ -0,0 +1,603 @@
+from abc import abstractmethod
+from operator import attrgetter
+
+import pydot
+
+from ..base import Base, Property
+from .node import Node, SensorNode, RepeaterNode, FusionNode
+from .edge import Edges, DataPiece, Edge
+from ..types.groundtruth import GroundTruthPath
+from ..types.detection import TrueDetection, Clutter
+from ._functions import _default_label
+
+from typing import List, Tuple, Collection, Set, Union, Dict
+import numpy as np
+import networkx as nx
+import graphviz
+from datetime import datetime, timedelta
+
+
+class Architecture(Base):
+    """Abstract Architecture Base class. Subclasses must implement the
+    :meth:`~Architecture.propogate` method.
+    """
+
+    edges: Edges = Property(
+        doc="An Edges object containing all edges. For A to be connected to B we would have an "
+            "Edge with edge_pair=(A, B) in this object.")
+    current_time: datetime = Property(
+        default=datetime.now(),
```

Comment:
To be replaced in the `__init__` with `datetime.now()`? It would seem inconvenient to me otherwise to have to specify a start, when this is unimportant to the majority of likely use cases.